src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(health, name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(health, name) {
+    super(health, name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(someAnimal) {
+    if (someAnimal.hidden !== true && someAnimal instanceof Herbivore) {
+      someAnimal.health -= 50;
+
+      if (someAnimal.health <= 0) {
+        delete Animal.alive.someAnimal;
+        Animal.alive.length--;
```

Comment:
is it necessary to decrease length here?

Reply:
yes its necessary because if I deletedsome element after that remains undefined and I need cut that undefined using length--  something like that